### ISSUE TITLE
Objects API: add API to get object metadata

### ISSUE THREAD

**Issue by talSofer** [treeverse/lakeFS#1981]
Currently, there is no API to get object metadata. Object metadata can be added for example on staging objects (https://github.com/treeverse/lakeFS/blob/b399c5ae482f8f93f51cb41c5c2f02ac36598e0e/clients/java/src/main/java/io/lakefs/clients/api/model/ObjectStageCreation.java#L53).

Implementation suggestions: 
- Modify the ObjectsApi.statObject to return object metadata. 
- Add a new API call to return the object metadata.

**Comment by adunuthulan** [comment]
Hey @talSofer! I'd like to take a look at this issue. 

I wanted to clarify the full definition of object metadata. This endpoint in the api 

`/repositories/{repository}/refs/{ref}/objects/stat:` 

says it "gets the object metadata", but I believe this doesn't refer to the user defined metadata assigned at Object Stage Creation. This endpoint corresponds to the StatObject function, which returns ObjectStats that hold every metadata information referenced in the `ObjectStageCreation.java` file other than user defined data:

```
ObjectStats{
		Checksum:        entry.Checksum,
		Mtime:           entry.CreationDate.Unix(),
		Path:            params.Path,
		PathType:        entryTypeObject,
		PhysicalAddress: qk.Format(),
		SizeBytes:       Int64Ptr(entry.Size),
}
```

So the issue is that there is no way of obtaining this user defined metadata through the API. Is my understanding correct?

**Comment by arielshaqed** [comment]
> So the issue is that there is no way of obtaining this user defined metadata through the API. Is my understanding correct?

Precisely!

**Comment by adunuthulan** [comment]
I'm confused about the difference between the `LinkPhysicalAddress()` and `StageObject()` actions. Both seem to have very similar implementations and both claim to `"stage_object"` for their logs. The response code of linking the physical address is **OK**, while for staging the response code is **Created**, but both operations create a DBEntry with the object metadata and staging doesn't do any extra creation. 

The description of the `LinkPhysicalAddress()` action in the API states that 
```
If the supplied token matches the current staging token, associate the object as the physical address with the 
supplied path.

Otherwise, if staging has been committed and the token has expired, return a conflict and hint where to place 
the object to try again. Caller should copy the object to the new physical address and PUT again with the new 
staging token. (No need to back off, this is due to losing the race against a concurrent commit operation.)
```

but the function implementation doesn't make use of the token supplied in the request body. It seems like these operations either can be merged, or that `LinkPhysicalAddress()` is missing some functionality that makes them different.

**Comment by talSofer** [comment]
@adunuthulan you are right both actions are almost identical. We have this open issue to unify them https://github.com/treeverse/lakeFS/issues/1866.
